jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -268,7 +268,19 @@ private static StringBuilder encodePath(StringBuilder buf, String path, int offs
 
         return buf;
     }
-    
+
+    /**
+     * Encode a raw URI String and convert any raw spaces to
+     * their "%20" equivalent.
+     *
+     * @param str input raw string
+     * @return output with spaces converted to "%20"
+     */
+    public static String encodeSpaces(String str)
+    {
+        return StringUtil.replace(str, " ", "%20");
```

Comment:
I'm wondering if there is benefit in having specialised methods for replacing char with string and string with char?

Reply:
That kind of pattern is usually highly specialized and probably needs a specialized method to remain consistent over all uses of it.

Take the new `StringUtil.sanitizeFileSystemPath()` as an example.

https://github.com/eclipse/jetty.project/blob/245388210311e3fce7c4300ec9c848ca01fc9637/jetty-util/src/main/java/org/eclipse/jetty/util/StringUtil.java#L143-L178